src/Shared/PackageManagers/MavenProjectManager.cs

```diff
@@ -213,12 +314,25 @@ public override async Task<bool> PackageVersionExistsAsync(PackageURL purl, bool
             }
             try
             {
-                string packageNamespace = purl.Namespace.Replace('.', '/');
                 string packageName = purl.Name;
-                string feedUrl = (purl?.Qualifiers?["repository_url"] ?? ENV_MAVEN_ENDPOINT).EnsureTrailingSlash();
+                string feedUrl = purl?.Qualifiers?["repository_url"] ?? ENV_MAVEN_ENDPOINT.GetRepositoryUrl();
+                MavenSupportedUpstream upstream = feedUrl.GetMavenSupportedUpstream();
+
                 HttpClient httpClient = CreateHttpClient();
 
-                string? content = await GetHttpStringCache(httpClient, $"{feedUrl}{packageNamespace}/{packageName}/{purl.Version}/", useCache);
+                string packageRepositoryCheckUri = string.Empty;
+                if (upstream == MavenSupportedUpstream.MavenCentralRepository)
+                {
+                    string packageNamespace = purl.Namespace.Replace('.', '/');
+                    packageRepositoryCheckUri = $"{upstream.GetRepositoryUrl().EnsureTrailingSlash()}{packageNamespace}/{packageName}/{purl.Version}/";
+                }
+                else if (upstream == MavenSupportedUpstream.GoogleMavenRepository)
+                {
+                    string packageNamespace = purl.Namespace;
+                    packageRepositoryCheckUri = $"{upstream.GetRepositoryUrl()}{packageNamespace}:{packageName}:{purl.Version}";
+                }
+
+                string? content = await GetHttpStringCache(httpClient, packageRepositoryCheckUri, useCache);
```

Comment:
Is there a more standard way we could check this? What if we sent a HEAD request for the `pom.xml` file that we expect to exist?